starknet_py/net/udc_deployer/deployer.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from typing import Optional, List, Union, cast, NamedTuple
+
+from starkware.starknet.definitions.fields import ContractAddressSalt
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.common import int_from_hex
+from starknet_py.constants import DEFAULT_DEPLOYER_ADDRESS
+from starknet_py.net.client_models import Hash, Call
+from starknet_py.net.models import AddressRepresentation, parse_address, compute_address
+from starknet_py.utils.contructor_args_translator import translate_constructor_args
+from starknet_py.utils.crypto.facade import pedersen_hash
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+)
+from starknet_py.utils.sync import add_sync_methods
+
+
+ContractDeployment = NamedTuple("ContractDeployment", [("udc", Call), ("address", int)])
+
+
+@add_sync_methods
+class Deployer:
+    """
+    Deployer used to deploy contracts through Universal Deployer Contract (UDC)
+    """
+
+    def __init__(
+        self,
+        *,
+        deployer_address: AddressRepresentation = DEFAULT_DEPLOYER_ADDRESS,
+        account_address: Optional[AddressRepresentation] = None,
```

Comment:
Maybe we should move it to `create_deployment_call`?

Reply:
It depends if we want the UDC instance to be reusable for different account addresses or created per address: I think in standard use case, making all deployments from the same account is more probable.

Reply:
I agree with @cptartur

Reply:
It's a bit weird from user perspective, we should be able to create the UDC with a minimal set of required properties. In this case `account_address` is a separate concept from a UDC.

Reply:
`account_address` can be used (with unique=True) to influence the address of the deployed contract, I though standard use case of the udc class instance would be configuring this parameter once and then making different deployments. 

We can move it to `create_deployment_call` if you think it makes more sense there, but for me it'd be more ergonomic.

Reply:
My logic is that we want to expose logic of function from UDC here, so we can expose same parameters (in our case unique becomes account_address).

Reply:
Still, I think it should be as reusable as possible. In most cases, users will be using this with only one account and passing the same `account_address` will be annoying

Reply:
I don't think this will be very annoying. It is not something that you do often and if you do you can always create a wrapper for it.

Reply:
You can also argue that if someone (for whatever reason) want's to reuse the same UDC instance for different addresses they can do `udc.account_address = new_address`.

If we make this module too directly modelled after UDC interface what will be the benefit of using it instead of just `Contract` aside from address calculation.

Reply:
I don't really se the usacase where someone want's to make multiple deploy calls through the same udc but with different addresses - they'd have to then go through different AccountClients anyway.

Reply:
>You can also argue that if someone (for whatever reason) want's to reuse the same UDC instance for different addresses they can do udc.account_address = new_address.

No, never do that 😆 

Ok, I don't have a strong opinion on this. It can stay in constructor.

Reply:
I don't recommend doing that, but if it's available as a public parameter anyways and for performance (or whatever) reasons someone cannot create multiple UDC objects, they might as well do that.

Reply:
No, just never do that :)